tests/test_mark_tokens.py

```diff
@@ -338,6 +339,39 @@ def test_adjacent_joined_strings(self):
             "JoinedStr:'x y z'   # comment2\n       'a b c'   # comment3\n       f'u v w'"
         })
 
+  if sys.version_info >= (3, 12):
+    def test_fstrings_3_12_plus(self):
+        m = self.create_mark_checker(
+          'x = (f"Wobble {f"{func(kwarg=f"{boo!r}")}"!r}.",)',
+        )
+
+        self.assertEqual(m.view_nodes_at(1, 6), {
+            'JoinedStr:f"Wobble {f"{func(kwarg=f"{boo!r}")}"!r}."',
+        })
+
+        m.verify_all_nodes(self)
+
+  def test_bytes_smoke(self):
+    const = 'Const' if self.is_astroid_test else (
+        'Constant'
+        if sys.version_info >= (3, 8) or six.PY2
+        else 'Bytes'
+    )
+
+    for source in (
+      'b"123abd"',
+      r"b'\x12\x3a\xbc'",
+    ):
+      expected = {"Module:" + source}
+      if six.PY3 or not self.is_astroid_test:
```

Comment:
It sounds like this is just broken in Python 2 with astroid. We don't have to fix that, but I think it'd be better to just skip this whole test in Python 2, remove the `six.PY2` further above, and collapse this `if` to make things readable. We're supposed to be dropping Python 2.

Reply:
I agree the conditionals here aren't great. Note however that dropping support for Python 2 doesn't change the shape of things much -- we still need all the conditions which are present in this code (the conditions just get a little bit simpler). If support for Python < 3.8 goes away too then that'd reduce this to just the astroid differences (in both parts).

Reply:
Surely `six.PY3 or not self.is_astroid_test` is always true in Python 3?

Reply:
:facepalm: I somehow wasn't reading the `not`.

Reply:
f422904b237b30157d9f04e3ccad8275aa65e9eb